Speed up DrawTriangles

seebs opened the pull request:
DrawTriangles is expensive and slow because of massive memory
allocation and garbage collection costs. This patch moves from ~47TPS
on my laptop (with ~24k triangles) to 60TPS. The first part
is just allocating the right size of vertex buffer up front; that
got to about 55TPS. The second part replaces the frequent
allocations of []float32 in Vertex() calls with writing the
desired values into a provided destination slice.

Time spent in drawing triangles for 1,000 frames:
	13.07s	baseline
	11.09s	preallocate whole buffer to avoid resizing
	6.13s	use new PutVertex function

This might need some cleanup, but I think it's good evidence that
the design change is viable.

hajimehoshi commented on internal/graphicsutil/vertices.go:
```diff
@@ -136,6 +136,33 @@ func QuadIndices() []uint16 {
+	vs[9] = ca
+}
+
 func Vertex(width, height int, dx, dy, sx, sy float32, cr, cg, cb, ca float32) []float32 {
```
I think `Vertex` is no longer used? :-)

hajimehoshi replied:
My suggestion is to remove `Vertex` and rename `PutVertex` to `Vertex`.

seebs commented:
I've updated this and rebased to master. I decided to keep the PutVertex name because the behavior is otherwise-surprising; most of the time, functions like this return things, rather than pushing them into a provided slice.

hajimehoshi commented:
I might change the name or signature, but `PutVertex` is ok :-)
